src/Microsoft.AspNet.Diagnostics/DeveloperExceptionPage/DeveloperExceptionPageExtensions.cs

```diff
@@ -31,7 +31,30 @@ public static IApplicationBuilder UseErrorPage([NotNull] this IApplicationBuilde
         /// <returns></returns>
         public static IApplicationBuilder UseErrorPage([NotNull] this IApplicationBuilder builder, ErrorPageOptions options)
         {
-            return builder.UseMiddleware<ErrorPageMiddleware>(options);
+            return builder.UseMiddleware<DeveloperExceptionPageMiddleware>(options);
+        }
+
+        /// <summary>
+        /// Captures synchronous and asynchronous exceptions from the pipeline and generates HTML error responses.
+        /// Full error details are only displayed by default if 'host.AppMode' is set to 'development' in the IApplicationBuilder.Properties.
+        /// </summary>
+        /// <param name="builder"></param>
+        /// <returns></returns>
+        public static IApplicationBuilder UseDeveloperExceptionPage([NotNull] this IApplicationBuilder builder)
```

Comment:
I am following the approach @Eilon suggested. Adding these methods in addition to existing ones. Will remove them once I finish cross repo changes and check them in.